frontend/packages/dev-console/src/components/edit-application/EditApplication.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react';
+import { Formik } from 'formik';
+import * as _ from 'lodash';
+import { NormalizedBuilderImages, normalizeBuilderImages } from '../../utils/imagestream-utils';
+import { createResources } from '../import/import-submit-utils';
+import { validationSchema } from '../import/import-validation-utils';
+import EditApplicationForm from './EditApplicationForm';
+import { EditApplicationProps } from './edit-application-types';
+import * as EditApplicationUtils from './edit-application-utils';
+
+const EditApplication: React.FC<EditApplicationProps> = ({
+  namespace,
+  appName,
+  resource,
+  resources,
+  onCancel,
+  onSubmit,
+}) => {
+  const builderImages: NormalizedBuilderImages =
+    resources.imageStreams && resources.imageStreams.loaded
+      ? normalizeBuilderImages(resources.imageStreams.data)
+      : null;
+
+  const currentImage = _.split(
+    _.get(resources, 'buildConfig.data.spec.strategy.sourceStrategy.from.name', ''),
+    ':',
+  );
+
+  const appGroupName = _.get(resource, 'metadata.labels["app.kubernetes.io/part-of"]');
+
+  const initialValues = {
+    formType: 'edit',
+    name: appName,
+    application: {
+      name: appGroupName,
+      selectedKey: appGroupName,
+    },
+    project: {
+      name: namespace,
+    },
+    git: EditApplicationUtils.getGitData(_.get(resources, 'buildConfig.data')),
+    docker: {
+      dockerfilePath: _.get(
+        resources,
+        'buildConfig.data.spec.strategy.dockerStrategy.dockerfilePath',
+        'Dockerfile',
+      ),
+      containerPort: parseInt(
+        _.split(_.get(resources, 'route.data.spec.port.targetPort'), '-')[0],
+        10,
+      ),
+    },
+    image: {
+      selected: currentImage[0] || '',
+      recommended: '',
+      tag: currentImage[1] || '',
+      tagObj: {},
+      ports: [],
+      isRecommending: false,
+      couldNotRecommend: false,
+    },
+    route: EditApplicationUtils.getRouteData(_.get(resources, 'route.data'), resource),
+    resources: EditApplicationUtils.getResourcesType(resource),
+    serverless: EditApplicationUtils.getServerlessData(resource),
+    pipeline: {
+      enabled: false,
+    },
+    build: EditApplicationUtils.getBuildData(_.get(resources, 'buildConfig.data')),
+    deployment: EditApplicationUtils.getDeploymentData(resource),
+    labels: EditApplicationUtils.getUserLabels(resource),
+    limits: EditApplicationUtils.getLimitsData(resource),
+  };
+
+  const handleSubmit = (values, actions) => {
+    const imageStream = builderImages && builderImages[values.image.selected].obj;
+
+    createResources(values, imageStream, false, false, resources, resource)
```

Comment:
This is confusing to me. Sending `resources` as well as `resource`. You canrename `resources` -> `appResources` as your tyep is also `AppResources`. Also you can add `originalResource` or `deploymentResource` to the `appResources` and send only one object.